### ISSUE TITLE
Requirements?

### ISSUE THREAD

**Issue by BeZazz** [nuBuilder/nuBuilder-4.5#14]
Hi all :)
No matter what I do (re/create delete change password) I keep getting 
`Connection failed: SQLSTATE[HY000] [1045] Access denied for user `

I am staring to wonder if it comes down to using versions of PHP and MariaDB that are not supported.

I am using PHP 7.4.27 and MariaDB 10.2.43

**Comment by smalos** [comment]
Hi,

Did you grant privileges on the nubuilder database for the user?

https://docs.phpmyadmin.net/en/latest/privileges.html

**Comment by BeZazz** [comment]
Thank you for the reply. Yes, I have granted and ungranted. Created different users, different databases. Removed and reuploaded files.

I realize the error I have is common/basic but I have tried everything I could think of. The control panel I use adds an "_" to the database and user names. Was going to see if I can get around that and see if it helps.

I have been through mod_sec logs, Apache etc to no avail. I was hoping it may of been a version issue.

**Comment by smalos** [comment]
Just to rule this out:  After making changes to nuconfig.php, the browser must be restarted in order for changes to be reflected.

What you could also do is create a small php file to test the db connection  independently of nuBuilder:


```
<?php

// ***** Update variables! **********
$servername = "localhost";
$username = "root";
$password = "";
$db = "nubuilder";
// ***********************************

try {
  $conn = new PDO("mysql:host=$servername;dbname=$db", $username, $password);
  $conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
  echo "Connected successfully";
} catch(PDOException $e) {
  echo "Connection failed: " . $e->getMessage();
}

?>
```

Save in in nuBuilder's root dir with the name, e.g., connection_test.php

**Comment by BeZazz** [comment]
I created the file and the output was
Connected successfully

**Comment by smalos** [comment]
> Just to rule this out: After making changes to nuconfig.php, the browser must be restarted in order for changes to be reflected.

Did you do that?

Next, I would add some debug code (right after the line $DBCharset = ....)  in nudatabase. php to inspect the db variables:

`echo "mysql:host=$DBHost;dbname=$DBName;charset=$DBCharset, $DBUser, $DBPassword";`

**Comment by BeZazz** [comment]
Yes computer have been restarted at least twice. 

Output
mysql:host=;dbname=;charset=, ,

**Comment by smalos** [comment]
That is really strange and it means that the db variables that are declared in nuconfig.php cannot be retrieved in nudatabase.php.

As a workaround, replace this block:

```
$DBHost					= $_SESSION['nubuilder_session_data']['DB_HOST'];
$DBName				= $_SESSION['nubuilder_session_data']['DB_NAME'];
$DBUser					= $_SESSION['nubuilder_session_data']['DB_USER'];
$DBPassword				= $_SESSION['nubuilder_session_data']['DB_PASSWORD'];
```
By adding your db connection details:

```
$DBHost					= 'your host...';
$DBName				= 'db name...';
$DBUser					= 'db user...';
$DBPassword				= 'db password...';
```

**Comment by BeZazz** [comment]
That worked to load up the login page but can not login. I imagine if I transfer over the login details somewhere something else will popup?

Any idea where I could start looking for the issue (Finding where/why they are disappearing)? 

Thanks for your help. I appreciate it.

**Comment by smalos** [comment]
Check the developer console for errors by clicking the `F12` key on most browsers.
Also check the (apache) server logs for any errors that might be related to that issue.

**Comment by BeZazz** [comment]
OK, will do. 
Thanks for your help.

(will reply when I have an outcome)

**Comment by BeZazz** [comment]
I got this working. Technically I made no changes.

I went through changing session options (restarting Apache every change) in php.ini, it started working when I enabled 
```
; Initialize session on request startup.
; http://php.net/session.auto-start
session.auto_start = 0 
```
However after setting it back to 0 it continued to work. I did a diff on a recent php.ini and a week old php.ini comparing to the current one after rebooting the server.
All three were the same. 

The uploaded files on the server are the same as what I downloaded from git. Apart from password, usernames added. 

So I made changes to php.ini one at a time. On the fifth Apache restart, it started working and going back to the original settings made no difference and the site continued to work.

thanks for all your help.
